src/pages/home.tsx

```diff
@@ -28,52 +32,52 @@ export const HomePage: React.FC = () => {
     const userVersion = localStorage.getItem("version");
 
     if (userVersion && userVersion !== latestVersion) {
-      document.getElementById("markdown")?.classList.remove("hidden");
+      if (dialog) {
+        dialog.show();
+      }
     }
     localStorage.setItem("version", latestVersion);
   }, []); // Run this effect only once, when the component mounts
 
   return (
-    <main>
-      <aside className="text-right">
-        <button
-          className="bg-cyan-900 text-right ml-auto mt-2 py-1 px-3 rounded-md"
-          onClick={() =>
-            document.getElementById("markdown")?.classList.remove("hidden")
-          }
-        >
-          View Update Notes
-        </button>
-      </aside>
-
-      <section className="max-w-full">
-        {recentEvents?.map((event) => (
-          <LinkButton
-            to={`/${event.sku}`}
-            className="w-full max-w-full mt-4"
-            key={event.sku}
+    <>
+      <main>
+        <aside className="text-right">
+          <button
+            className="bg-cyan-900 text-right ml-auto mt-2 py-1 px-3 rounded-md"
+            onClick={() => dialog.show()}
           >
-            <p className="text-sm">
-              <span className=" text-emerald-400 font-mono">{event.sku}</span>
-            </p>
-            <p className="">{event.name}</p>
-          </LinkButton>
-        ))}
-      </section>
-      <div
-        className="hidden absolute top-8 left-1/2 rounded-md p-4 bg-neutral-800 -translate-x-1/2 w-11/12"
-        id="markdown"
+            View Update Notes
+          </button>
+        </aside>
+
+        <section className="max-w-full">
+          {recentEvents?.map((event) => (
+            <LinkButton
+              to={`/${event.sku}`}
+              className="w-full max-w-full mt-4"
+              key={event.sku}
+            >
+              <p className="text-sm">
+                <span className=" text-emerald-400 font-mono">{event.sku}</span>
+              </p>
+              <p className="">{event.name}</p>
+            </LinkButton>
+          ))}
+        </section>
+      </main>
+      <dialog
```

Comment:
Moved the dialog outside the <main> as that makes more sense

Reply:
Looks good. I wrote a wrapper on top of the default `<dialog>` element in html, `<Dialog>` that you may want to use. It would probably look something like.

```typescript

const [updateDialogOpen, setUpdateDialogOpen] = useState(false);

 useEffect(() => {
    // Fetch the update notes as a markdown file
    const fetchMarkdownContent = async () => {
      try {
        const response = await fetch("/updateNotes.md");
        const content = await response.text();
        setMarkdownContent(content);
      } catch (error) {}
    const userVersion = localStorage.getItem("version");

    if (userVersion && userVersion !== latestVersion) {
      setUpdateDialogOpen(true)
    }
    localStorage.setItem("version", latestVersion);
  }, []); 
``

```typescript
 <Dialog
        open={updateDialogOpen}
        mode={DialogMode.Modal}
        onClose={() => setUpdateDialogOpen(false)}
      >
        <DialogHeader title="What's New" onClose={() => setUpdateDialogOpen(false)} />
        <DialogBody>
         <Markdown content={markdownContent) />
       </DialogBody>
</Dialog>
```

Reply:
Thanks, that makes the scrolling behaviour much easier, and makes the dialog consistent with other selection screens.